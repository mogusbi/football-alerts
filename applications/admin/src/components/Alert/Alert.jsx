import Button from '@material-ui/core/Button';
import Dialog from '@material-ui/core/Dialog';
import DialogActions from '@material-ui/core/DialogActions';
import DialogContent from '@material-ui/core/DialogContent';
import DialogTitle from '@material-ui/core/DialogTitle';
import List from '@material-ui/core/List';
import ListItem from '@material-ui/core/ListItem';
import ListItemText from '@material-ui/core/ListItemText';
import PropTypes from 'prop-types';
import React, {memo} from 'react';

function Alert ({closeHandler, message, open, title}) {
  return (
    <Dialog
      aria-labelledby='alert-dialog-title'
      onClose={closeHandler}
      open={open}
    >
      <DialogTitle id='alert-dialog-title'>
        {title}
      </DialogTitle>
      <DialogContent>
        <List dense={false}>
          {
            message.map((item) => (
              <ListItem key={item}>
                <ListItemText>
                  {item}
                </ListItemText>
              </ListItem>
            ))
          }
        </List>
      </DialogContent>
      <DialogActions>
        <Button
          color='primary'
          onClick={closeHandler}
        >
          Close
        </Button>
      </DialogActions>
    </Dialog>
  );
}

Alert.propTypes = {
  closeHandler: PropTypes.func.isRequired,
  message: PropTypes.arrayOf(PropTypes.string).isRequired,
  open: PropTypes.bool.isRequired,
  title: PropTypes.string.isRequired
};

export default memo(Alert);
