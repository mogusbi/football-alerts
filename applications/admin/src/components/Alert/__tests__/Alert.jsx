import Button from '@material-ui/core/Button';
import {mount, shallow} from 'enzyme';
import React from 'react';
import Alert from '../Alert';

describe('Alert', () => {
  let closeHandler;
  let message;
  let open;
  let title;
  let wrapper;

  beforeEach(() => {
    closeHandler = jest.fn();
    title = 'Test alert';
    message = [
      'This is the first message',
      'This is the second message'
    ];
  });

  describe('when open', () => {
    beforeEach(() => {
      open = true;

      wrapper = shallow(
        <Alert
          closeHandler={closeHandler}
          message={message}
          open={open}
          title={title}
        />
      );
    });

    it('should render component', () => {
      expect(wrapper.dive()).toMatchSnapshot();
    });

    it('should call closeHandler when closed', () => {
      const closeButton = wrapper.find(Button);

      closeButton.simulate('click');

      expect(closeHandler).toHaveBeenCalledTimes(1);
    });
  });

  describe('when closed', () => {
    beforeEach(() => {
      open = false;

      wrapper = mount(
        <Alert
          closeHandler={closeHandler}
          message={message}
          open={open}
          title={title}
        />
      );
    });

    it('should render component', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
