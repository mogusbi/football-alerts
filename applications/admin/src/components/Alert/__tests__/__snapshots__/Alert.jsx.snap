// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert when closed should render component 1`] = `
<Alert
  closeHandler={[MockFunction]}
  message={
    Array [
      "This is the first message",
      "This is the second message",
    ]
  }
  open={false}
  title="Test alert"
>
  <WithStyles(Dialog)
    aria-labelledby="alert-dialog-title"
    onClose={[MockFunction]}
    open={false}
  >
    <Dialog
      PaperComponent={[Function]}
      TransitionComponent={[Function]}
      aria-labelledby="alert-dialog-title"
      classes={
        Object {
          "container": "MuiDialog-container-4",
          "paper": "MuiDialog-paper-5",
          "paperFullScreen": "MuiDialog-paperFullScreen-14",
          "paperFullWidth": "MuiDialog-paperFullWidth-13",
          "paperScrollBody": "MuiDialog-paperScrollBody-7",
          "paperScrollPaper": "MuiDialog-paperScrollPaper-6",
          "paperWidthLg": "MuiDialog-paperWidthLg-11",
          "paperWidthMd": "MuiDialog-paperWidthMd-10",
          "paperWidthSm": "MuiDialog-paperWidthSm-9",
          "paperWidthXl": "MuiDialog-paperWidthXl-12",
          "paperWidthXs": "MuiDialog-paperWidthXs-8",
          "root": "MuiDialog-root-1",
          "scrollBody": "MuiDialog-scrollBody-3",
          "scrollPaper": "MuiDialog-scrollPaper-2",
        }
      }
      disableBackdropClick={false}
      disableEscapeKeyDown={false}
      fullScreen={false}
      fullWidth={false}
      maxWidth="sm"
      onClose={[MockFunction]}
      open={false}
      scroll="paper"
      transitionDuration={
        Object {
          "enter": 225,
          "exit": 195,
        }
      }
    >
      <WithStyles(Modal)
        BackdropProps={
          Object {
            "transitionDuration": Object {
              "enter": 225,
              "exit": 195,
            },
          }
        }
        aria-labelledby="alert-dialog-title"
        className="MuiDialog-root-1"
        closeAfterTransition={true}
        disableBackdropClick={false}
        disableEscapeKeyDown={false}
        onClose={[MockFunction]}
        open={false}
        role="dialog"
      >
        <Modal
          BackdropComponent={[Function]}
          BackdropProps={
            Object {
              "transitionDuration": Object {
                "enter": 225,
                "exit": 195,
              },
            }
          }
          aria-labelledby="alert-dialog-title"
          className="MuiDialog-root-1"
          classes={
            Object {
              "hidden": "MuiModal-hidden-16",
              "root": "MuiModal-root-15",
            }
          }
          closeAfterTransition={true}
          disableAutoFocus={false}
          disableBackdropClick={false}
          disableEnforceFocus={false}
          disableEscapeKeyDown={false}
          disablePortal={false}
          disableRestoreFocus={false}
          hideBackdrop={false}
          keepMounted={false}
          manager={
            ModalManager {
              "data": Array [],
              "handleContainerOverflow": true,
              "hideSiblingNodes": true,
              "modals": Array [],
            }
          }
          onClose={[MockFunction]}
          open={false}
          role="dialog"
        />
      </WithStyles(Modal)>
    </Dialog>
  </WithStyles(Dialog)>
</Alert>
`;

exports[`Alert when open should render component 1`] = `
<Dialog
  PaperComponent={[Function]}
  TransitionComponent={[Function]}
  aria-labelledby="alert-dialog-title"
  classes={
    Object {
      "container": "MuiDialog-container-4",
      "paper": "MuiDialog-paper-5",
      "paperFullScreen": "MuiDialog-paperFullScreen-14",
      "paperFullWidth": "MuiDialog-paperFullWidth-13",
      "paperScrollBody": "MuiDialog-paperScrollBody-7",
      "paperScrollPaper": "MuiDialog-paperScrollPaper-6",
      "paperWidthLg": "MuiDialog-paperWidthLg-11",
      "paperWidthMd": "MuiDialog-paperWidthMd-10",
      "paperWidthSm": "MuiDialog-paperWidthSm-9",
      "paperWidthXl": "MuiDialog-paperWidthXl-12",
      "paperWidthXs": "MuiDialog-paperWidthXs-8",
      "root": "MuiDialog-root-1",
      "scrollBody": "MuiDialog-scrollBody-3",
      "scrollPaper": "MuiDialog-scrollPaper-2",
    }
  }
  disableBackdropClick={false}
  disableEscapeKeyDown={false}
  fullScreen={false}
  fullWidth={false}
  maxWidth="sm"
  onClose={[MockFunction]}
  open={true}
  scroll="paper"
  transitionDuration={
    Object {
      "enter": 225,
      "exit": 195,
    }
  }
>
  <WithStyles(DialogTitle)
    id="alert-dialog-title"
  >
    Test alert
  </WithStyles(DialogTitle)>
  <WithStyles(DialogContent)>
    <WithStyles(List)
      dense={false}
    >
      <WithStyles(ListItem)
        key="This is the first message"
      >
        <WithStyles(ListItemText)>
          This is the first message
        </WithStyles(ListItemText)>
      </WithStyles(ListItem)>
      <WithStyles(ListItem)
        key="This is the second message"
      >
        <WithStyles(ListItemText)>
          This is the second message
        </WithStyles(ListItemText)>
      </WithStyles(ListItem)>
    </WithStyles(List)>
  </WithStyles(DialogContent)>
  <WithStyles(DialogActions)>
    <WithStyles(Button)
      color="primary"
      onClick={[MockFunction]}
    >
      Close
    </WithStyles(Button)>
  </WithStyles(DialogActions)>
</Dialog>
`;
